src/pages/index.js

```diff
@@ -103,70 +116,176 @@ const settings = {
 const editFormElement = editProfileModal.querySelector(".modal__form");
 const addFormElement = addCardModal.querySelector(".modal__form");
 
+const avatarModalPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  (avatarData) => {
+    avatarModalPopup.setButtonText(true);
+    api
+      .editUserPhoto(avatarData)
+      .then((data) => {
+        userInfo.setAvatar(data.avatar);
+        avatarModalPopup.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => avatarModalPopup.setButtonText());
+  }
+);
+
+const avatarModalElement = avatarModal.querySelector(".modal__form");
+
+/*----------------------------------*/
+/*      card like/remove likes      */
+/*----------------------------------*/
+
+// const handleLikeCard = (card) => {
+//   if (card._isliked) {
+//     api
+//       .removeCardLike(card._cardId)
+//       .then(() => {
+//         card.setLikes(false);
+//       })
+//       .catch((err) => {
+//         console.error(err);
+//       });
+//   } else {
+//     api
+//       .addCardLike(card._cardId)
+//       .then(() => {
+//         card.setLikes(true);
+//       })
+//       .catch((err) => {
+//         console.error(err);
+//       });
+//   }
+// };
+
 /*----------------------*/
 /*      popup items     */
 /*----------------------*/
 
-const editFormValidator = new FormValidator(settings, editFormElement);
-const addFormValidator = new FormValidator(settings, addFormElement);
-
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
-
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
-);
+const addCardPopup = new PopupWithForm("#add-card-modal", (cardData) => {
+  addCardPopup.setButtonText(true);
+  return api
+    .addCard(cardData)
+    .then((res) => {
+      console.log(res);
+      createCard(res);
+    })
+    .catch((error) => {
+      console.error(error);
+    });
+});
 addCardPopup.setEventListeners();
 
-const editProfilePopup = new PopupWithForm(
-  "#edit-modal",
-  handleProfileFormSubmit
-);
+const editProfilePopup = new PopupWithForm("#edit-modal", (card) => {
+  editProfilePopup.setButtonText(true);
```

Comment:
**Could be improved**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **closing** and **catching errors**


```js

// You can make a universal function that accepts a request, popup instance and optional loading text  
function handleSubmit(request, popupInstance, loadingText = "Saving...") {

  // here we change the button text
  popupInstance.renderLoading(true, loadingText);
  request()
    .then(() => {
      // We need to close only in `then`
      popupInstance.close()
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
     .catch(console.error)
    // in `finally` we need to return the initial button text back in any case
    .finally(() => {
      popupInstance.renderLoading(false);
    });
}



```


You need to pass the **request** into the call of `handleSubmit` like that:



```js

// here is an example of the profile form handling
function handleProfileFormSubmit(inputValues) {
  // we create a function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally` inside `handleSubmit`
    return api.editProfile(inputValues).then((userData) => {
      userInfo.setUserInfo(userData)
    });
  }
  // Here we call the function passing the request, popup instance and if we need some other loading text we can pass it as the 3rd argument
  handleSubmit(makeRequest, profilePopup);
}



```